### DISCUSSION TITLE
module dependency version requirements

### DISCUSSION THREAD

**Discussion by 0cyn** [binref/refinery#44] [Q&A] [answered]
Hey! First off, I love this project. 

I'm the author of [ktool](https://github.com/0cyn/ktool) which machometa entirely relies upon, and recently pushed a new version to pypi with 2 years of changes. I tried to make sure all of the code within this still works with the updated version, but I also made several improvements to it I wanted to PR here while the old methods are in their 'deprecated' state, and these changes wont work against the old 1.3 version if users still have it installed.  

Is there a way for me to mark a new ref of machometa as requiring that 2.0 version of the library?

**Comment by huettenhain** [comment, marked as answer]
Hey there, and first of all, thank you so much! This is incredibly forthcoming and very kind. The [dependency to k2l is coded into the unit itself](https://github.com/binref/refinery/blob/56a2806666e004d21eca870180378c1b385ac25e/refinery/units/formats/macho/machometa.py#L47), and you can alter the code to add a version requirement. For example, this would require a `k2l` version at least 2.0:
```python
    @Unit.Requires('k2l>=2.0', 'all')
    def _ktool():
        import ktool
        import ktool.macho
        import ktool.codesign
        return ktool
```
I hope this answers the question, please let me know if I completely missed the point 😅.

**Comment by jhhcs** [comment]
To expand on this: Refinery does by default just install the most recent version, and I had to impose the negation of that version requirement in https://github.com/binref/refinery/commit/bb7ca554767d0b6b28fff2dae97b76e10571762a so that the current code doesn't break 😅.
